src/features/multichain/components/CreateSafeOnNewChain/index.tsx

```diff
@@ -69,27 +69,33 @@ const ReplaySafeDialog = ({
   const customRpc = useAppSelector(selectRpc)
   const dispatch = useAppDispatch()
   const [creationError, setCreationError] = useState<Error>()
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState<boolean>(false)
 
   // Load some data
   const [safeCreationData, safeCreationDataError, safeCreationDataLoading] = safeCreationResult
 
   const onFormSubmit = handleSubmit(async (data) => {
+    setIsSubmitDisabled(true)
+
     const selectedChain = chain ?? replayableChains?.find((config) => config.chainId === data.chainId)
     if (!safeCreationData || !selectedChain) {
+      setIsSubmitDisabled(false)
       return
     }
 
     // We need to create a readOnly provider of the deployed chain
     const customRpcUrl = selectedChain ? customRpc?.[selectedChain.chainId] : undefined
     const provider = createWeb3ReadOnly(selectedChain, customRpcUrl)
     if (!provider) {
+      setIsSubmitDisabled(false)
       return
     }
 
     // 1. Double check that the creation Data will lead to the correct address
     const predictedAddress = await predictAddressBasedOnReplayData(safeCreationData, provider)
     if (!sameAddress(safeAddress, predictedAddress)) {
       setCreationError(new Error('The replayed Safe leads to an unexpected address'))
+      setIsSubmitDisabled(false)
       return
```

Comment:
Should we just use a try-catch-finally where we `setIsSubmitDisabled(false)` in the finally block?